lib/core/src/swapper/boltz/client.rs

```diff
@@ -0,0 +1,142 @@
+use crate::{
+    bitcoin, elements,
+    model::{BlockchainExplorer, Config},
+};
+use boltz_client::{
+    error::Error,
+    network::{
+        electrum::{ElectrumBitcoinClient, ElectrumLiquidClient},
+        esplora::{EsploraBitcoinClient, EsploraLiquidClient},
+        BitcoinChain, BitcoinClient as BoltzBitcoinClient, LiquidChain,
+        LiquidClient as BoltzLiquidClient,
+    },
+};
+use sdk_macros::async_trait;
+
+const BOLTZ_CONNECTION_TIMEOUT: u8 = 100;
+
+pub(crate) enum LiquidClient {
+    Electrum(Box<ElectrumLiquidClient>),
+    Esplora(Box<EsploraLiquidClient>),
+}
+
+impl LiquidClient {
+    pub(crate) fn new(config: &Config) -> Result<Self, Error> {
+        let (tls, validate_domain) = config.tls_settings();
+        Ok(match &config.liquid_explorer {
+            BlockchainExplorer::Electrum { url } => {
+                Self::Electrum(Box::new(ElectrumLiquidClient::new(
+                    config.network.into(),
+                    url,
+                    tls,
+                    validate_domain,
+                    BOLTZ_CONNECTION_TIMEOUT,
+                )?))
+            }
+            BlockchainExplorer::Esplora { url, .. } => {
+                Self::Esplora(Box::new(EsploraLiquidClient::new(
+                    config.network.into(),
+                    url,
+                    BOLTZ_CONNECTION_TIMEOUT as u64,
+                )))
+            }
+        })
+    }
+}
+
+#[async_trait]
```

Comment:
To make it wasm friendly
```suggestion
#[sdk_macros::async_trait]
```

Reply:
It's using the `sdk_macros` trait, just implicitly. I'll set it to explicit

Reply:
Ah sorry, I missed it 🙃